imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -141,37 +144,69 @@ def update(self) -> None:
         t_begin = self.get_current_time()
         self.mf6.prepare_time_step(0.0)
 
-        self.delt = self.mf6.get_time_step()
-        self.msw.prepare_time_step(self.delt)
+        self.dtgw = self.mf6.get_time_step()
+        self.dtsw = self.msw.get_sw_time_step()
+        self.number_substeps_per_modflowstep = int(self.dtgw / self.dtsw)
+        self.msw.prepare_time_step(self.dtgw)
+
+        # initialise water balance
+        self.exchange_balans.initialise()
 
         # stage from dflow 1d to modflow active coupled riv
         self.exchange_stage_1d_dfm2mf6()
-        # flux from modflow active coupled riv to dflow 1d
+
+        # flux from modflow active coupled riv to water balance
         self.exchange_flux_riv_active_mf62dfm()
 
-        # flux from modflow passive coupled riv to dflow 1d
+        # flux from modflow passive coupled riv to water balance
         self.exchange_flux_riv_passive_mf62dfm()
 
-        # flux from modflow passive coupled drn to dflow 1d
+        # flux from modflow passive coupled drn to water balance
         self.exchange_flux_drn_passive_mf62dfm()
 
-        # get cum flux mf->fm pre-timestep and store locally
-        self.store_1d_river_fluxes_to_dfm()
-
         # sub timestepping between metaswap and dflow
         subtimestep_endtime = t_begin
-        for _ in range(self.number_dflowsteps_per_modflowstep):
-            subtimestep_endtime += self.delt / self.number_dflowsteps_per_modflowstep
+        for idtsw in range(self.number_substeps_per_modflowstep):
+            subtimestep_endtime += self.dtsw
+            self.msw.start_surface_water_time_step(idtsw)
+            # flux from metaswap ponding to water balance
+            self.exchange_ponding_msw2dflow1d()
+
+            # flux from metaswap sprinkling to water balance
+            self.exchange_sprinkling_msw2dflow1d()
+
+            # exchange water balance to dlfow
+            self.exchange_balans.sum_demand()
+            self.exchange_balans_2dflow(self.exchange_balans.demand["sum"])
 
+            # get cummelative flux before dfm-run
+            q_dflow0 = self.dfm.get_cumulative_fluxes_1d_nodes()
+
+            # run dflow
             while (
                 self.dfm.get_current_time()
                 < days_to_seconds(subtimestep_endtime) - self.time_eps
             ):
                 self.dfm.update()
```

Comment:
check my response in branch "iMOD6-1020". I think the timesteps are small enough, but i'm not sure as well maybe this can be a separate task to investigate.